src/controllers/getArticle.js

```
@@ -8,15 +17,27 @@ const getArticle = async (request, reply) => {
 
   if (!article) return reply.code(404).send({ error: `Failed to find article id ${articleId}` });
 
-  let { publishHistory } = article;
+  let { publishHistory, alternateLang } = article;
   const { uri } = publishHistory;
 
   if (!uri.endsWith('-flyway')) {
     publishHistory = { ...publishHistory, uri: `${uri}-flyway` };
   }
 
+  if (alternateLang && alternateLang.length > 0) {
+    try {
+      alternateLang = alternateLang.map((alternateId) => {
+        const toBeStubbed = brandArticles.find(singleArticle => singleArticle.id === alternateId);
+        if (!toBeStubbed) throw new Error(`Bad article alternateLang mapping for article. (Article: ${articleId}. NestedId: ${alternateId})`);
+        return createStubFromArticle(toBeStubbed, brand, request.origin);
+      });
+    } catch (error) {
+      return reply.code(500).send({ error: error.message });
```

I don't think you need the try/catch block at all. I believe this is the behaviour of fastly if you just let the error be thrown (which will become a Promise rejection).